src/sentry/nodestore/base.py

```diff
@@ -85,3 +94,44 @@ def generate_id(self):
 
     def cleanup(self, cutoff_timestamp):
         raise NotImplementedError
+
+    def _get_cache_item(self, id):
+        if self.cache:
+            return self.cache.get(id)
+
+    def _get_cache_items(self, id_list):
+        if self.cache:
+            return self.cache.get_many(id_list)
+        return {}
+
+    def _set_cache_item(self, id, data):
+        if self.cache and data:
+            if random.random() < self.sample_rate:
+                self.cache.set(id, data)
```

Comment:
How are we managing the TTL ? Is it configured elsewhere ?

Reply:
We have default TTLs set in configuration. If we wanted to provide user-configured TTLs, it's easier to set it at the cache definition level than to bring in the notion of configuring Nodestore itself.

Reply:
1h as per https://github.com/getsentry/getsentry/pull/3484 (private repo)